src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/MSBuildProject.File.cs

```diff
@@ -158,6 +158,41 @@ public void RenameFile(string filePath)
         public void AddItem(string name, string path)
             => ProjectRoot.AddItem(name, path);
 
+        public bool RemoveItem(ProjectItemType? itemType = null, string? includePath = null, string? excludePath = null, string? elementName = null)
+        {
+            var items = ProjectRoot.Items.Where(item =>
+            {
+                if (itemType != null && item.ItemType != itemType.Name)
+                {
+                    return false;
+                }
+
+                if (includePath != null && item.Include != includePath)
+                {
+                    return false;
+                }
+
+                if (excludePath != null && item.Exclude != excludePath)
+                {
+                    return false;
+                }
+
+                if (elementName != null && item.ElementName != elementName)
+                {
+                    return false;
+                }
+
+                return true;
+            });
+
+            foreach (var item in items)
+            {
+                item.Parent.RemoveChild(item);
+            }
+
```

Comment:
The assumption with the `RemoveItem` is to remove a item ideally a leaf node. Have you tested this with removing a nested item ie. x -> y -> z and you are trying to remove y, what happens then?

Reply:
For now this removes everything that fit the parameters including something that potentially has a child. 
Are you recommending that we only delete an item if it is a leaf node?